pkg/tsdb/influxdb/fsql/fsql.go

```diff
@@ -97,6 +97,12 @@ func runnerFromDataSource(dsInfo *models.DatasourceInfo) (*runner, error) {
 		addr += ":443"
 	}
 
+	// If the user has specified an address with no scheme it can still be valid
+	// So we use the raw URL value
+	if u.Host == "" {
```

Comment:
is this supposed to be `if u.Scheme == "" {`

Reply:
This is the correct check. If the user passes a URL without a scheme (e.g. `mydomain:8181`) then `url.Parse` will parse the URL with the scheme `mydomain` rather than the host.

You can see this on the Go playground by running:

```go
package main

import (
	"fmt"
	"net/url"
)

func main() {
	u, _ := url.Parse("mydomain:8181")
	fmt.Printf("host: %s \n", u.Host)
	fmt.Printf("scheme: %s \n", u.Scheme)
	fmt.Printf("port: %s \n", u.Port())
}
```

Great point on adding a separate function for this parsing to unit test it.